osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -204,17 +206,25 @@ protected override void Update()
 
             if (lastFetchCompleted)
             {
-                APIRequest nextPageRequest = null;
+                if (ScorePanelList.IsEmpty)
+                {
+                    // This can happen if a beatmap part of a playlist hasn't been played yet.
+                    VerticalScrollContent.Add(new MessagePlaceholder(LeaderboardStrings.NoRecordsYet));
```

Comment:
This adds _one `MessagePlaceholder` to the drawable hierarchy every frame_ if there are no scores.

Leave the results screen open for long enough and you will run out of system memory.